src/org/labkey/test/components/core/DetailComponentTable.java

```diff
@@ -0,0 +1,106 @@
+package org.labkey.test.components.core;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Wraps an oft-used table that uses data-fieldKey attributes to display key/column pairs
+ */
+public class DetailComponentTable extends WebDriverComponent<DetailComponentTable.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public DetailComponentTable(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public Map<String, String> getData()
+    {
+        Map data = new HashMap();
+        List<String> keys = elementCache().fieldKeys();
+        for (String key : keys)
+            data.put(key, elementCache().getTableValue(key));
+        return data;
+    }
+
+    public String getTableValue(String fieldKey)
+    {
+        return elementCache().getTableValue(fieldKey);
+    }
+
+    public List<String> fieldKeys()
+    {
+        return elementCache().fieldKeys();
+    }
+
```

Comment:
Should add a getter for the Component name. The h4 tag with class attribute of lineage-name-data.

Reply:
This is implemented in a function on the LineageGraph class; the name div (along with the overview and lineage links and its icon) is outside of the DetailComponentTable's componentElement.

LineageGraph now has the following getters for the current/selected node:
getFocusedNodeData() - gets a map of the contents of the detailComponentTable
getFocusedNodeImage() - returns the icon/image
getFocusedNodeText() - returns the text of the h4.lineage-name-data element